compiler/src/iree/compiler/API/Internal/CompilerDriver.cpp

```diff
@@ -90,6 +89,74 @@
 namespace mlir::iree_compiler::embed {
 namespace {
 
+// If note using memfd_create, then we need to align output buffers
+// similarly, which is unfortunately, quite platform specific.
+// While memfd_create aligns to a page, we align these to 64 bytes,
+// which matches runtime requirements.
+const size_t kOutputBufferAlignment = 64;
+
+template <typename T>
+struct rt_aligned_allocator {
+  using value_type = T;
+  rt_aligned_allocator() noexcept {}
+  template <class U>
+  rt_aligned_allocator(const rt_aligned_allocator<U> &) noexcept {}
```

Comment:
Do we expect IREE to be ever built with exceptions enabled and benefit from `noexcept`?

Reply:
No. But allocators are finicky and this is the exact standard compliant way to define one.